astropy/coordinates/transformations.py

```diff
@@ -43,30 +45,40 @@ def __init__(self):
         self._graph = defaultdict(dict)
         self.invalidate_cache()  # generates cache entries
 
-    @property
+    @lazyproperty
     def _cached_names(self):
-        if self._cached_names_dct is None:
-            self._cached_names_dct = dct = {}
-            for c in self.frame_set:
-                nm = getattr(c, 'name', None)
-                if nm is not None:
-                    dct[nm] = c
+        cached_names = {}
+        for c in self.frame_set:
+            nm = getattr(c, 'name', None)
+            if nm is not None:
+                cached_names[nm] = c
 
-        return self._cached_names_dct
+        return cached_names
 
-    @property
+    @lazyproperty
```

Comment:
The change to `lazyproperty` here and below aren't really performance gains per se, but the code does look a bit nicer.